packages/core/content-manager/admin/src/pages/EditView/components/DocumentActions.tsx

```diff
@@ -788,16 +892,46 @@ const UpdateAction: DocumentActionComponent = ({
   const { create, update, clone, isLoading } = useDocumentActions();
   const [{ rawQuery }] = useQueryParams();
   const onPreview = usePreviewContext('UpdateAction', (state) => state.onPreview, false);
+  const { getInitialFormValues } = useDoc();
 
   const isSubmitting = useForm('UpdateAction', ({ isSubmitting }) => isSubmitting);
   const modified = useForm('UpdateAction', ({ modified }) => modified);
   const setSubmitting = useForm('UpdateAction', ({ setSubmitting }) => setSubmitting);
   const document = useForm('UpdateAction', ({ values }) => values);
   const validate = useForm('UpdateAction', (state) => state.validate);
   const setErrors = useForm('UpdateAction', (state) => state.setErrors);
-  const resetForm = useForm('PublishAction', ({ resetForm }) => resetForm);
+  const resetForm = useForm('UpdateAction', ({ resetForm }) => resetForm);
+
+  const dispatch = useRelationModal('UpdateAction', (state) => state.dispatch);
+
+  // need to discriminate if the update is coming from a relation modal or in the edit view
+  const relationContext = useRelationModal('UpdateAction', () => true, false);
+  const fieldToConnect = useRelationModal(
+    'UpdateAction',
+    (state) => state.state.fieldToConnect,
+    false
+  );
+  const documentHistory = useRelationModal(
+    'UpdateAction',
+    (state) => state.state.documentHistory,
+    false
+  );
+  const rootDocumentMeta = useRelationModal('UpdateAction', (state) => state.rootDocumentMeta);
+  const fromRelationModal = relationContext != undefined;
 
   const { currentDocumentMeta } = useDocumentContext('UpdateAction');
+  const [updateDocumentMutation] = useUpdateDocumentMutation();
```

Comment:
Why do we do this directly instead of getting`update` from `useDocumentActions`? That's where we get the publish mutation as well on line 540. The isLoading destructured as well and that includes loading state for publish || update. Not sure if that is beneficial or not but just fyi

Reply:
We did it in this way with Remi because otherwise we would have Two notification success messages for the create and the update

Reply:
ok what about the loading part, should the loading state returned in PublishAction consider the loading state of this mutation? See line 818

Reply:
for sure we have an issue with loading and we can open a GH issue for that, and refactor a little bit this code, before merging the future branch

Reply:
>We did it in this way with Remi because otherwise we would have Two notification success messages for the create and the update

also otherwise it would send 2 tracking events for the same action

Reply:
I get that we have issues to resolve around loading in the relation modal. Just want to clarify my point that here you are introducing a loading state that is not handled. It is not an existing issue. For example the publish button in the relation modal is only going to show a loader for the publish action, even though an update to the parent document could still be in progress after the publish is finished. 

Unless I'm missing something, and I'll leave it up to you if you want to handle it in another PR, but I would recommend going ahead and doing this in the relevant places:

```
const [updateDocumentMutation, { isLoading: isLoadingUpdate }] = useUpdateDocumentMutation();
// ....
return {
  loading: isLoading || isLoadingUpdate
}
```